src/main/java/edu/cmu/oli/content/boundary/managers/AnalyticsResourceManager.java

```diff
@@ -123,11 +97,135 @@ public JsonElement createDataset(AppSecurityContext session, final String packag
+        datasetInfo.addProperty("message", message);
         return datasetInfo;
     }
 
+    private List<Dataset> getProcessingDatasets(String packageGuid) {
+        TypedQuery<Dataset> q = em.createNamedQuery("Dataset.findProcessingByPackageGuid", Dataset.class);
+        q.setParameter("packageGuid", packageGuid);
+        return (List<Dataset>) q.getResultList();
+    }
+
+    // Three slices - byPart, bySkill, and byResource
+    private Map<String, String> datasetJsonToTsv(Dataset dataset) {
+        Map<String, String> output = new HashMap<>();
+        JsonObject json = dataset.getDatasetBlob().getBodyJson().getAsJsonObject();
+        output.put("byResource", byResourceToTsv(json.get("byResource").getAsJsonArray()));
+        output.put("byPart", byPartToTsv(json.get("byPart").getAsJsonArray()));
+        output.put("bySkill", bySkillToTsv(json.get("bySkill").getAsJsonArray()));
+        return output;
+    }
+
+    private String byResourceToTsv(JsonArray dataRows) {
+        List<String> titleHeaders = Arrays.asList("Resource", "Title");
+        Set<String> titleKeys = new LinkedHashSet<String>();
+        titleKeys.addAll(Arrays.asList("resource", "title"));
+
+        return datasetToTsv(titleHeaders, titleKeys, dataRows);
+    }
+
+    private String byPartToTsv(JsonArray dataRows) {
+        List<String> headers = Arrays.asList("ID", "Resource", "Title", "Revision", "Part", "Submit and Compare");
+        Set<String> keys = new LinkedHashSet<String>();
+        keys.addAll(Arrays.asList("id", "resourceId", "resourceTitle", "revision", "part", "submitAndCompare"));
+
+        return datasetToTsv(headers, keys, dataRows);
+    }
+
+    private String bySkillToTsv(JsonArray dataRows) {
+        List<String> headers = Arrays.asList("Skill", "Title");
+        Set<String> keys = new LinkedHashSet<String>();
+        keys.addAll(Arrays.asList("skill", "title"));
+
+        return datasetToTsv(headers, keys, dataRows);
+    }
+
+    private String datasetToTsv(List<String> headers, Set<String> keys, JsonArray dataRows) {
+        List<String> commonHeaders = Arrays.asList("Distinct Students", "Distinct Registrations", "Opportunities",
+                "Practice", "Hints", "Errors", "Eventually Correct", "First Response Correct", "Utilization - Start",
+                "Utilization - Finish", "Average Help Needed", "Average Number of Tries", "Completion Rate",
+                "Accuracy Rate");
+        List<String> allHeaders = new ArrayList<>();
+        allHeaders.addAll(headers);
+        allHeaders.addAll(commonHeaders);
+
+        List<String> commonKeys = Arrays.asList("distinctStudents", "distinctRegistrations", "opportunities",
+                "practice", "hints", "errors", "correct", "firstResponseCorrect", "utilizationStart",
+                "utilizationFinish", "avgHelpNeeded", "avgNumberOfTries", "completionRate", "accuracyRate");
+        Set<String> allKeys = new LinkedHashSet<>();
```

Comment:
This seems interesting.  Why a LinkedHashSet?  A comment explaining the motivation here would be helpful for future generations.